### DISCUSSION TITLE
How to remove a file or buffer from context?

### DISCUSSION THREAD

**Discussion by gavinhughes** [karthink/gptel#487] [Q&A] [answered]
From the `gptel-menu`, the `-b` and `-f` switches are used to add buffers and files to the context. Is there a direct way to 1) remove specific files or buffers that have been previously added, and/or 2) clear the context entirely?

For 2), maybe map the below to `c` in the gptel menu?

``` elisp
(defun gptel-clear-context ()
  "Prompts for confirmation before clearing the `gptel-context--alist`."
  (interactive)
  (when (y-or-n-p "Are you sure you want to clear the gptel context? ")
    (setq gptel-context--alist nil)
    (message "Context cleared.")))
```

**Comment by karthink** [comment, marked as answer]
For 2: yes, I think it's a good idea to add a command to dump the context.  There's a little more to it than resetting `gptel-context--alist`, see my response in #486.

For 1: You can remove specific context chunks from the context inspection buffer.  Press `d` to mark a context chunk for deletion, `n` and `p` to navigate, it's the same keybindings as dired/buffer-list/ibuffer:

![image](https://github.com/user-attachments/assets/4612b2c2-1978-4fc4-8f13-3de75a700353)

For regions or buffers, you can also clear context chunks from their destination locations, run `M-x gptel-add` again to toggle context.
